features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/messagecomposer/MessageComposerPresenter.kt

```diff
@@ -152,6 +152,10 @@ class MessageComposerPresenter @Inject constructor(
                     showAttachmentSourcePicker = false
                     cameraVideoPicker.launch()
                 }
+                MessageComposerEvents.PickAttachmentSource.Location -> {
+                    showAttachmentSourcePicker = false
+                    // Navigation to the location picker screen is done at the view layer
+                }
```

Comment:
I'm not sure what the answer is - and it's definitely out of scope for this PR - but I dislike the way some events bubble through handlers on the state, and some bubble up the view hierarchy. It feels like a leaky abstraction.

Reply:
I pointed that out too.
We could build a `Navigator` interface, whose the parent node will implement, and pass it to the presenter so we can call navigation actions from there.

I opted not to do it in this PR though.